src/ucp/proto/proto_common.inl

```diff
@@ -57,15 +57,18 @@ ucp_proto_request_zcopy_init(ucp_request_t *req, ucp_md_map_t md_map,
 
     ucp_proto_completion_init(&req->send.state.uct_comp, comp_func);
 
-    status = ucp_datatype_iter_mem_reg(ep->worker->context,
-                                       &req->send.state.dt_iter,
-                                       md_map, uct_reg_flags, dt_mask);
-    if (status != UCS_OK) {
-        return status;
+    /* rndv/put allready initalized iterator in RTS */
+    if (uct_reg_flags) {
```

Comment:
why needed? maybe assert it's !=0? or let ucp_datatype_iter_mem_reg handle flags==0?

Reply:
As in comment - it might already be initialized in RTS (if not fragmented)

Reply:
can we let ucp_datatype_iter_mem_reg() take care of it?